src/index.js

```diff
@@ -160,23 +160,23 @@ export default class Trigger extends React.Component {
         this.clickOutsideHandler = addEventListener(currentDocument,
           'mousedown', this.onDocumentClick);
       }
+      // close popup when trigger type is `contextMenu`, or `click` and window is blurred
+      if (!this.clickBlurOutsideHandler && (this.isClickToHide() || this.isContextMenuToShow())) {
+        this.clickBlurOutsideHandler = addEventListener(window,
+          'blur', this.onDocumentClick);
+      }
       // always hide on mobile
       if (!this.touchOutsideHandler) {
         currentDocument = currentDocument || props.getDocument();
         this.touchOutsideHandler = addEventListener(currentDocument,
           'touchstart', this.onDocumentClick);
       }
       // close popup when trigger type contains 'onContextMenu' and document is scrolling.
-      if (!this.contextMenuOutsideHandler1 && this.isContextMenuToShow()) {
+      if (!this.contextMenuScrollOutsideHandler && this.isContextMenuToShow()) {
         currentDocument = currentDocument || props.getDocument();
-        this.contextMenuOutsideHandler1 = addEventListener(currentDocument,
+        this.contextMenuScrollOutsideHandler = addEventListener(currentDocument,
```

Comment:
Sorry for delay. `contextMenuScrollOutsideHandler` should get remove after unmounted. And also need a test case for this.

Reply:
@stephen-james